wave_vision/src/fast_detector.cpp

```diff
@@ -0,0 +1,87 @@
+// Libwave Headers
+#include "wave/vision/fast_detector.hpp"
+
+namespace wave {
+
+FASTDetector::FASTDetector(const std::string &config_path) {
+    // Extract parameters from .yaml file.
+    ConfigParser parser;
+
+    // Configuration parameters
+    int param_threshold;
+    bool param_nonmax_suppression;
+    int param_type;
+
+    // Add parameters to parser, to be loaded. If path cannot be found, throw an
+    // exception.
+    parser.addParam("fast.threshold", &param_threshold);
+    parser.addParam("fast.nonmax_suppression", &param_nonmax_suppression);
+    parser.addParam("fast.type", &param_type);
+
+    if (parser.load(config_path) != 0) {
+        ConfigurationLoadingException exception;
+        throw exception;
+    }
+
+    // Verify configuration values
+    checkConfiguration(param_threshold, param_type);
+
+    // Create FastFeatureDetector with these parameters
+    this->fast_detector = cv::FastFeatureDetector::create(param_threshold,
+                                                          param_nonmax_suppression,
+                                                          param_type);
+}
+
+FASTDetector::~FASTDetector() {}
+
+void FASTDetector::checkConfiguration(int new_threshold, int new_type) {
+    // Create InvalidConfigurationException object
+    InvalidConfigurationException exception;
+
+    // Check parameters. If invalid, throw an exception.
+    if (new_threshold < 0) {
+        std::cout << "Invalid threshold for FASTDetector!" << std::endl;
+        throw exception;
+    } else if (new_type < 0 || new_type > 3) {
+        std::cout << "Invalid type for FASTDetector!" << std::endl;
+        throw exception;
+    }
+}
+
+void FASTDetector::configure(int new_threshold,
+                                     bool new_nonmax_suppression,
+                                     int new_type) {
+    // Confirm configuration parameters are valid
+    checkConfiguration(new_threshold, new_type);
+
+    // Store configuration values in member variables
```

Comment:
Configuration parameters are no longer a member variable, this concern should be addressed now.